### PR TITLE
Implementing Azure Application insight

### PR DESCRIPTION
**Pull request by warlof**
This is a draft proposal to switch from our home made Google Analytics to something which is much more designed to debug purpose.

image

image

### CODE REVIEW

**Comment by warlof** [comment]
For those who want to contribute to the testing bench, you can add the following at the end of your `.env` file :

`AZURE_APP_INSIGHT_KEY=71f57ad7-2777-45fd-ac6b-5ce1252269ee`

**Comment by herpaderpaldent** [comment]
I keep my critics: don't send exception stack to Analytics.

You should not use Analytics to DEBUG. Even if you would so you need to change the documentation because now you would violate the secrecy claim of @leonjza .

If you would make it possible to send Analytics without exception stack then it would be finde as you give users the choice to help with statistics without exposing sensitive intel.

**Comment by stcktrce** [comment]
Have you considered adding support for https://sentry.io/welcome/ for stack traces instead? There is a click-and-play Laravel library, and it's pretty easy to filter out sensitive information from stack traces.

This might resolve some of the issues pointed out by @herpaderpaldent

File: app/Exceptions/Handler.php [hunk 1]
```diff
@@ -39,6 +40,8 @@ public function report(Exception $exception)
     {
 
         parent::report($exception);
+
```
Comment (herpaderpaldent) [inline app/Exceptions/Handler.php]: add global settings `allow_sending_exception_stack`

override exception if ` setting('allow_sending_exception_stack', false);`
``` 
$exception = new EmptyException;
```

File: app/Wrappers/AzureApplicationInsight.php [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        if (! $this->beforeTrackTelemetry())
+            return;
+
+        // build a name composed of HTTP verb and request path
+        $name = sprintf('%s %s', strtoupper($request->method()), $request->path());
+
+        // build an extra properties array
+        $properties = [
+            'ajax' => $request->ajax(),
+            'ip' => $request->ip(),
```
Comment (herpaderpaldent) [inline app/Wrappers/AzureApplicationInsight.php]: This must be anonymized: as of today and to be in line with the intend of Google Analytics and documentation about analytics

https://github.com/eveseat/services/blob/master/src/Jobs/Analytics.php#L173

**Pull request closed without merging**